src/conduits/RwaInputConduit3.sol

```diff
@@ -0,0 +1,227 @@
+     * @param _to RwaUrn contract address.
+     * @param _quitTo Address to where outstanding GEM balance will go after `quit`
+     */
+    constructor(
+        address _psm,
+        address _to,
+        address _quitTo
+    ) public {
+        DSTokenAbstract _gem = DSTokenAbstract(GemJoinAbstract(PsmAbstract(_psm).gemJoin()).gem());
+        psm = PsmAbstract(_psm);
+        dai = DSTokenAbstract(PsmAbstract(_psm).dai());
+        gem = _gem;
+        to = _to;
+        quitTo = _quitTo;
+
+        // Give unlimited approve to PSM gemjoin
+        _gem.approve(address(PsmAbstract(_psm).gemJoin()), type(uint256).max);
+
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
+    }
+
+    /*//////////////////////////////////
+               Authorization
+    //////////////////////////////////*/
+
+    /**
+     * @notice Grants `usr` admin access to this contract.
+     * @param usr The user address.
+     */
+    function rely(address usr) external auth {
+        wards[usr] = 1;
+        emit Rely(usr);
+    }
+
+    /**
+     * @notice Revokes `usr` admin access from this contract.
+     * @param usr The user address.
+     */
+    function deny(address usr) external auth {
+        wards[usr] = 0;
+        emit Deny(usr);
+    }
+
+    /**
+     * @notice Grants `usr` push access to this contract.
+     * @param usr The user address.
+     */
+    function mate(address usr) external auth {
+        may[usr] = 1;
+        emit Mate(usr);
+    }
+
+    /**
+     * @notice Revokes `usr` push access from this contract.
+     * @param usr The user address.
+     */
+    function hate(address usr) external auth {
+        may[usr] = 0;
+        emit Hate(usr);
+    }
+
+    /*//////////////////////////////////
+               Administration
+    //////////////////////////////////*/
+
+    /**
+     * @notice Updates a contract parameter.
+     * @param what The changed parameter name. `"quitTo", "to"`
+     * @param data The new value of the parameter.
+     */
+    function file(bytes32 what, address data) external auth {
+        if (what == "quitTo") {
+            require(data != address(0), "RwaInputConduit3/invalid-quit-to-address");
+            quitTo = data;
+        } else if (what == "to") {
+            require(data != address(0), "RwaInputConduit3/invalid-to-address");
+            to = data;
+        } else {
+            revert("RwaInputConduit3/unrecognised-param");
+        }
+
+        emit File(what, data);
+    }
+
+    /*//////////////////////////////////
+               Operations
+    //////////////////////////////////*/
+
+    /**
+     * @notice Method to swap USDC contract balance to DAI through PSM and push it into RwaUrn address.
+     * @dev `msg.sender` must first receive push acess through mate().
+     */
+    function push() external isMate {
+        uint256 balance = gem.balanceOf(address(this));
+        require(balance > 0, "RwaInputConduit3/insufficient-gem-balance");
```

Comment:
I really think we shouldn't have this check here.
There's no effect of calling this function when it has no balance other than wasting gas.

If we leave it here, it will behave different than the other conduits, which don't revert when calling `push()` with no balance.

Reply:
Done